content/source/docs/enterprise/api/run.html.md

```diff
@@ -346,14 +428,66 @@ Parameter | Description
+The `cancel` action can be used to interrupt a run that is currently planning or applying. Performing a cancel is roughly equivalent to hitting ctrl+c during a Terraform plan or apply on the CLI. The running Terraform process is sent an `INT` signal, which instructs Terraform to end its work and wrap up in the safest way possible.
 
 This endpoint queues the request to perform a cancel; the cancel might not happen immediately. After canceling, the run is completed and later runs can proceed.
 
 This endpoint represents an action as opposed to a resource. As such, it does not return any object in the response body.
 
 -> **Note:** This endpoint cannot be accessed with [organization tokens](../users-teams-organizations/service-accounts.html#organization-service-accounts). You must access it with a [user token](../users-teams-organizations/users.html#api-tokens) or [team token](../users-teams-organizations/service-accounts.html#team-service-accounts).
 
+Status  | Response                  | Reason(s)
+--------|---------------------------|----------
+[202][] | none                      | Successfully queued a cancel request.
+[409][] | [JSON API error object][] | Run was not planning or applying, has not been canceled non-forcefully, or the cool-off period has not yet passed.
+
+### Request Body
+
+This POST endpoint allows an optional JSON object with the following properties as a request payload.
+
+Key path  | Type   | Default | Description
+----------|--------|---------|------------
+`comment` | string | `null`  | An optional explanation for why the run was canceled.
+
+### Sample Payload
+
+This payload is optional, so the `curl` command will work without the `--data @payload.json` option too.
+
+```json
+{
+  "comment": "This run was stuck and would never finish."
+}
+```
+
+### Sample Request
+
+```shell
+curl \
+  --header "Authorization: Bearer $TOKEN" \
+  --header "Content-Type: application/vnd.api+json" \
+  --request POST \
+  --data @payload.json \
+  https://app.terraform.io/api/v2/runs/run-DQGdmrWMX8z9yWQB/actions/cancel
+```
+
+## Forcefully cancel a Run
+
+`POST /runs/:run_id/actions/force-cancel`
+
+Parameter | Description
+----------|------------
+`run_id`  | The run ID to cancel
+
+The `force-cancel` action is like [cancel](#cancel-a-run), but ends the run immediately. Once invoked, the run is placed into a `canceled` state, and the running Terraform process is terminated. The workspace is immediately unlocked, allowing further runs to be queued.
+
+This endpoint enforces a prerequisite that a [non-forceful cancel](#cancel-a-run) is performed first. Once a non-forceful cancel has been performed, an additional cool-off period is enforced. The time remaining in the cool-off period can be found using the [run details endpoint](#get-run-details), in the key `force_cancel_delay_seconds`. Once this time period expires, if the run has still not ended, the force-cancel endpoint may be used successfully.
```

Comment:
User-brain questions: 

* What's the full path to this key? Is it `data.attributes.force_cancel_delay_seconds`, or is it inside a sub-object of `attributes`?
* I don't see the delay key in the sample run objects; does that mean it's only present in the run object while the cooldown delay is in effect? 
* Is the client meant to use the `permissions` and `actions` objects to determine whether force-cancel is an option? How does that work, and does it interact with the cooldown timer?

Reply:
Good questions, and I can try to clarify in the language here in the docs!

* Your full path is correct (`data.attributes.force_cancel_delay_seconds`).
* The key is indeed only present in the run output if the run has been canceled previously. If the run has not yet been canceled, there is no cool-off timer in the payload. After cool-off has expired the value will remain `0` after that.
* Permissions and actions can totally be used to figure out whether your user is allowed to force cancel, and if the run is in a state that would allow that to be successful. I can add that to the docs to clarify.

Reply:
🎉 thanks!